### PR TITLE
feature/ZEB-1010 implement saving launcher config

### PR DESCRIPTION
**Pull request by MitskevichVS**

### CODE REVIEW

File: client/app/shared/ci-helper/ci-helper.controller.js [hunk 1]
```diff
@@ -1054,47 +1134,49 @@ const CiHelperController = function CiHelperController(
...
-                    if (rs.data && rs.data.length) {
-                        $scope.scmAccounts = rs.data.filter(function (scmAccount) {
-                            return scmAccount.repositoryURL;
-                        });
+            });
+        getTenantInfo()
+            .then(tenant => {
+                isMultitenant = tenant.multitenant;
+                getClientId()
+                    .then(clientId => {
```
Comment (DmitryMishin) [inline client/app/shared/ci-helper/ci-helper.controller.js:1153]: You can use the following way to remove nesting:
```
getTenantInfo()
  .then(tenant => {
    isMultitenant = tenant.multitenant;
    return getClientId();
  })
  .then(clientId => $scope.clientId = clientId);
```

File: client/app/shared/ci-helper/ci-helper.controller.js [hunk 2]
```diff
@@ -1054,47 +1134,49 @@ const CiHelperController = function CiHelperController(
...
+                        $scope.clientId = clientId;
+                    });
+            });
+        const scmAccountsPromise = ScmService.getAllScmAccounts()
+            .then(function (rs) {
+                //TODO: why do we use $q here?
+                return $q(function (resolve, reject) {
```
Comment (DmitryMishin) [inline client/app/shared/ci-helper/ci-helper.controller.js:1160]: I'm not sure you need $q

File: client/app/shared/ci-helper/ci-helper.controller.js [hunk 3]
```diff
@@ -1054,47 +1134,49 @@ const CiHelperController = function CiHelperController(
...
+                });
             });
-        });
         const providersConfigPromise = getProvidersConfig();
 
         $q.all([launchersPromise, scmAccountsPromise, providersConfigPromise])
-            .then(function (data) {
-                $scope.scmAccounts.forEach(function (scmAccount) {
-                    scmAccount.launchers = $scope.launchers.filter(({ scmAccountType }) => scmAccountType.id === scmAccount.id);
+            .then(data =>{
```
Comment (DmitryMishin) [inline client/app/shared/ci-helper/ci-helper.controller.js:1177]: space after arrow

File: client/app/shared/ci-helper/ci-helper.controller.js [hunk 4]
```diff
@@ -1054,47 +1134,49 @@ const CiHelperController = function CiHelperController(
...
+                $scope.scmAccounts.forEach(scmAccount => {
+                    scmAccount.launchers = vm.launchers.filter(({ scmAccountType }) => scmAccountType.id === scmAccount.id);
```
Comment (DmitryMishin) [inline client/app/shared/ci-helper/ci-helper.controller.js:1179]: mutation is bad)

File: client/app/shared/ci-helper/ci-helper.controller.js [hunk 5]
```diff
@@ -1109,40 +1191,43 @@ const CiHelperController = function CiHelperController(
...
-                if (Array.isArray(platform.job) && platform.job.includes(type)) {
-                    vm.platformModel[vm.platformsConfig.rootKey] = platform;
-                    onPlatformSelect();
+            const type = (/-web$/i).test(vm.activeLauncher.type) ? 'web' : vm.activeLauncher.type;
 
-                    return true;
-                }
-
-                return false;
+            preselectedPlatform =  vm.platforms.find(platform => {
```
Comment (DmitryMishin) [inline client/app/shared/ci-helper/ci-helper.controller.js:1214]: double space

File: client/app/shared/ci-helper/ci-helper.controller.js [hunk 6]
```diff
@@ -1443,12 +1587,17 @@ const CiHelperController = function CiHelperController(
...
+        if (!vm.failedProvider && provider.data) {
+            initPlatforms(provider.data);
+        }
+        prepareLauncherControls();
     }
 
     function handleProviderDeselection() {
         clearPlatforms();
         vm.chipsCtrl && (vm.chipsCtrl.selectedChip = -1);
+        prepareLauncherControls()
```
Comment (DmitryMishin) [inline client/app/shared/ci-helper/ci-helper.controller.js:1600]: semicolon

File: client/app/shared/ci-helper/ci-helper.html [hunk 7]
```diff
@@ -34,15 +34,30 @@
                     </li>
                     <li class="folder-container_item">
                         <ul class="folder-container_item_list">
-                            <li class="folder-container_item_list_item hide-phone launcher-{{launcher.id}}" ng-click="chooseLauncher(launcher)" ng-repeat="launcher in scmAccount.launchers | orderBy: ['!preference.favorite', 'name'] | filter: filerSc track by launcher.id">
+                            <li class="folder-container_item_list_item hide-phone launcher-{{launcher.id}}"
+                                ng-class="{'_no-padding-bottom': $ctrl.activeLauncherId === launcher.id && launcher.presets.length}"
```
Comment (DmitryMishin) [inline client/app/shared/ci-helper/ci-helper.html:38]: wrong codestyle

File: client/app/shared/ci-helper/ci-helper.controller.js [hunk 8]
```diff
@@ -494,27 +552,28 @@ const CiHelperController = function CiHelperController(
 
     $scope.createLauncher = function (launcher) {
         return $q(function (resolve, reject) {
```
Comment (DmitryMishin) [inline client/app/shared/ci-helper/ci-helper.controller.js:554]: it's better to return LauncherService

**Review by dzmitrymishyn — APPROVED** [review]

File: client/app/shared/ci-helper/ci-helper.controller.js (hunk 6 above)
Comment (MitskevichVS) [reply client/app/shared/ci-helper/ci-helper.controller.js]: fixed

File: client/app/shared/ci-helper/ci-helper.controller.js (hunk 8 above)
Comment (MitskevichVS) [reply client/app/shared/ci-helper/ci-helper.controller.js]: fixed

File: client/app/shared/ci-helper/ci-helper.controller.js (hunk 1 above)
Comment (MitskevichVS) [reply client/app/shared/ci-helper/ci-helper.controller.js]: done

File: client/app/shared/ci-helper/ci-helper.controller.js (hunk 2 above)
Comment (MitskevichVS) [reply client/app/shared/ci-helper/ci-helper.controller.js]: fixed

File: client/app/shared/ci-helper/ci-helper.controller.js (hunk 3 above)
Comment (MitskevichVS) [reply client/app/shared/ci-helper/ci-helper.controller.js]: fixed

File: client/app/shared/ci-helper/ci-helper.controller.js (hunk 5 above)
Comment (MitskevichVS) [reply client/app/shared/ci-helper/ci-helper.controller.js]: fixed

File: client/app/shared/ci-helper/ci-helper.html (hunk 7 above)
Comment (MitskevichVS) [reply client/app/shared/ci-helper/ci-helper.html:38]: nope

**Pull request merged**